### PR TITLE
feat: Edit automation modal

### LINKED ISSUE #3576: feat: Add the ability to edit/delete/clone automation rules

### PR DESCRIPTION
**Pull request by fayazara**
## Description

Allows editing a exisitng automation

Fixes #3576 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Jest specs for vuex modules, api controllers and services.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/settings/automation/EditAutomationRule.vue [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+  </div>
+</template>
+
+<script>
+import alertMixin from 'shared/mixins/alertMixin';
+import { required, requiredIf } from 'vuelidate/lib/validators';
+import filterInputBox from 'dashboard/components/widgets/FilterInput.vue';
+import automationActionInput from 'dashboard/components/widgets/AutomationActionInput.vue';
+import languages from 'dashboard/components/widgets/conversation/advancedFilterItems/languages';
+import countries from '/app/javascript/shared/constants/countries.js';
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/automation/EditAutomationRule.vue:138]:
```suggestion
import countries from 'shared/constants/countries.js';
```

**Review by iamsivin — COMMENTED** [review]
@fayazara  Please take a look at the comments.

**Comment by fayazara** [comment]
> @fayazara Please take a look at the comments.

Fixed

**Review by iamsivin — DISMISSED** [review]
LGTM 👍🏻

File: app/javascript/dashboard/routes/dashboard/settings/automation/Index.vue [hunk 2]
```diff
@@ -54,7 +54,7 @@
                   @click="openEditPopup(automation)"
                 >
                 </woot-button>
-                <woot-button
+                <!-- s<woot-button
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/automation/Index.vue:57]:
```suggestion
                <!-- <woot-button
```

**Review by nithindavid — COMMENTED** [review]
Minor comments left

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**